auth.go

```diff
@@ -55,6 +55,40 @@ func Auth(update httprouter.Handle) httprouter.Handle {
 	}
 }
 
+// AuthUnregister middleware for unregister request
+func AuthUnregister(unregister httprouter.Handle) httprouter.Handle {
```

Comment:
I'm not very happy about implementing a new authentication middleware, as the core functionality stays the same. I do realize however that the code does need refactoring here and is suboptimal for many parts. I still think that they should be merged.

Another issue that I have is that AuthUnregister endpoint doesn't respect the IP based allowlist, which I think it should. Even though I see the use cases where it would be nice to be able to omit this check (like losing control over an IP address, and wanting to clear up the registrations for that), it's still a security option that the user chooses when registering an acme-dns account and I believe should be restricted in all the subsequent requests.